zipline/finance/performance/position_tracker.py

```diff
@@ -350,14 +350,17 @@ def get_positions(self):
                     pass
                 continue
 
-            # Note that this will create a position if we don't currently have
-            # an entry
+            # Returns the position if we have one, or a default position if we
+            # don't
             position = positions[sid]
             position.amount = pos.amount
             position.cost_basis = pos.cost_basis
             position.last_sale_price = pos.last_sale_price
             position.last_sale_date = pos.last_sale_date
 
+            # Adds the new position if we didn't have one before
+            positions[sid] = position
```

Comment:
it's a bit confusing to be getting or creating the position via `positions[sid]`.  I think it would be clearer to explicitly create a new position on line 355, assign it all the properties, then put it into `positions` via the code on line 362.